controllers/space/space_controller.go

```diff
@@ -312,6 +314,12 @@ func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, space *toolchainv1a
 	}
 }
 
+func setParentSpaceLabel(space *toolchainv1alpha1.Space) {
+	if parentSpace, found := space.Labels[toolchainv1alpha1.ParentSpaceLabelKey]; !found || parentSpace != space.Spec.ParentSpace {
+		space.Labels[toolchainv1alpha1.ParentSpaceLabelKey] = space.Spec.ParentSpace
```

Comment:
if the space is not a sub-space, which means that it doesn't have the `Space.Spec.ParentSpace` set, then the label shouldn't be present. However, this would set the label to empty string `""`

Reply:
Thanks! 

I've pushed a change to handle it and unit tests, please see: https://github.com/codeready-toolchain/host-operator/pull/727/commits/5ec86f07f3db70397e2dc2d60e356c5ccbc21415